stdlib.sh

```diff
@@ -986,6 +986,37 @@ layout_pipenv() {
   export VIRTUAL_ENV
 }
 
+# Usage: layout poetry
+#
+# Similar to layout_python, but uses Poetry [1] to build a virtualenv from the
+# pyproject.toml [2] located in the same directory.
+#
+# Poetry by default will create a virtual environment under
+# {cache-dir}/virtualenvs where cache-dir equates to $XDG_CACHE_HOME/pypoetry
+# or use the {project-dir}/.venv directory when one is available.
+#
+# [1]: https://python-poetry.org/
+# [2]: https://peps.python.org/pep-0518/
+#
+layout_poetry() {
+    PYPROJECT_TOML="${PYPROJECT_TOML:-pyproject.toml}"
+    if [[ ! -f "$PYPROJECT_TOML" ]]; then
+        log_error "No pyproject.toml found. Executing \`poetry init\` to create a \`$PYPROJECT_TOML\` first."
+        poetry init
```

Comment:
I included `poetry init` here as you queried whether such a command existed for `pipenv` in PR #314. This line could be improved further with something like the following:

```
# Specify the active python rather than the one used to install poetry in case they are not the same.
poetry init --python ^$(python3 --version 2>/dev/null | cut -d' ' -f2 | cut -d. -f1-2)
```
But I wasn't sure people would like it.